src/coreComponents/unitTests/physicsSolversTests/testWavePropagation.cpp

```diff
@@ -0,0 +1,227 @@
+  "    <FieldSpecification\n"
+  "      name=\"initialPressureN\"\n"
+  "      initialCondition=\"1\"\n"
+  "      setNames=\"{ all }\"\n"
+  "      objectPath=\"nodeManager\"\n"
+  "      fieldName=\"pressure_n\"\n"
+  "      scale=\"0.0\"/>\n"
+  "    <FieldSpecification\n"
+  "      name=\"initialPressureNm1\"\n"
+  "      initialCondition=\"1\"\n"
+  "      setNames=\"{ all }\"\n"
+  "      objectPath=\"nodeManager\"\n"
+  "      fieldName=\"pressure_nm1\"\n"
+  "      scale=\"0.0\"/>\n"
+  "    <FieldSpecification\n"
+  "      name=\"cellVelocity\"\n"
+  "      initialCondition=\"1\"\n"
+  "      objectPath=\"ElementRegions/Region/elementSubRegions/cb\"\n"
+  "      fieldName=\"mediumVelocity\"\n"
+  "      scale=\"1500\"\n"
+  "      setNames=\"{ all }\"/>\n"
+  "    <FieldSpecification\n"
+  "      name=\"zposFreeSurface\"\n"
+  "      objectPath=\"faceManager\"\n"
+  "      fieldName=\"FreeSurface\"\n"
+  "      scale=\"0.0\"\n"
+  "      setNames=\"{ zpos }\"/>\n"
+  "  </FieldSpecifications>\n"
+  "  <Tasks>\n"
+  "    <PackCollection\n"
+  "      name=\"waveFieldNp1Collection\"\n"
+  "      objectPath=\"nodeManager\"\n"
+  "      fieldName=\"pressure_np1\"/>\n"
+  "    <PackCollection\n"
+  "      name=\"waveFieldNCollection\"\n"
+  "      objectPath=\"nodeManager\"\n"
+  "      fieldName=\"pressure_n\"/>\n"
+  "    <PackCollection\n"
+  "      name=\"waveFieldNm1Collection\"\n"
+  "      objectPath=\"nodeManager\"\n"
+  "      fieldName=\"pressure_nm1\"/>\n"
+  "  </Tasks>\n"
+  "</Problem>\n";
+
+class AcousticWaveEquationSEMTest : public ::testing::Test
+{
+public:
+
+  AcousticWaveEquationSEMTest():
+    state( std::make_unique< CommandLineOptions >( g_commandLineOptions ) )
+  {}
+
+protected:
+
+  void SetUp() override
+  {
+    setupProblemFromXML( state.getProblemManager(), xmlInput );
+  }
+
+  static real64 constexpr time = 0.0;
+  static real64 constexpr dt = 1e-1;
+  static real64 constexpr eps = std::numeric_limits< real64 >::epsilon();
+
+  GeosxState state;
+  AcousticWaveEquationSEM * propagator;
+};
+
+real64 constexpr AcousticWaveEquationSEMTest::time;
+real64 constexpr AcousticWaveEquationSEMTest::dt;
+real64 constexpr AcousticWaveEquationSEMTest::eps;
+
+TEST_F( AcousticWaveEquationSEMTest, SeismoTrace )
+{
+
+  DomainPartition & domain = state.getProblemManager().getDomainPartition();
+  propagator = &state.getProblemManager().getPhysicsSolverManager().getGroup< AcousticWaveEquationSEM >( "acousticSolver" );
+  real64 time_n = time;
+  // run for 1s (10 steps)
+  for( int i=0; i<10; i++ )
+  {
+    propagator->solverStep( time_n, dt, i, domain );
+    time_n += dt;
+  }
+  // cleanup (triggers calculation of the remaining seismograms data points)
+  propagator->cleanup( 1.0, 10, 0, 0, domain );
+
+  // retrieve seismo
+  arrayView2d< real64 > const p_rcvs = propagator->getReference< array2d< real64 > >( AcousticWaveEquationSEM::viewKeyStruct::pressureNp1AtReceiversString() ).toView();
```

Comment:
```suggestion
  arrayView2d< real64 > const p_rcvs = propagator->getReference< array2d< real64 > >( AcousticWaveEquationSEM::viewKeyStruct::pressureNp1AtReceiversString() ).toView();
  p_rcvs.move( LvArray::MemorySpace::host, false );
```
to make sure that on the GPU platforms, the latest values of `p_rcvs` are available on the CPU when we do things like:
```
ASSERT_TRUE( std::abs( p_rcvs[i][8] ) > 0 );
```
I may be wrong, but I think this manual move is needed because `p_rcvs` is moved back to the CPU only if we go inside [this loop](https://github.com/GEOSX/GEOSX/blob/7664d3659f488d335fb37d94965623d881b90a9a/src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp#L335), which may not always be the case as it depends on the value of `m_nsamplesSeismoTrace` and `iSeismo`?

Reply:
I think we could consider moving the data from GPU to CPU only at the last iteration since we don't need it before. 
Also in our case using pygeosx, if we want to do some visualization during the simulation, we could imagine having a ".move()" if the variable lives on GPU, inside the routine of "get_wrapper('path_to_variable').value()" on C++ side. This could be useful not only for our variable but for any variable living on GPU.